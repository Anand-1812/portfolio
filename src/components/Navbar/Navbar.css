/* Navbar */
.navbar {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: var(--color-navbar);
  padding: 0.6rem 1.5rem;
  color: var(--color-text);
  transition: background-color var(--transition-speed), color var(--transition-speed);
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.05);
  /* subtle depth */
}

.navbar h1 {
  font-size: 1.4rem;
  /* smaller logo text */
  font-weight: 600;
}

.link {
  margin-left: 1rem;
}

.navbar ul {
  flex: 1;
  list-style: none;
  display: flex;
  justify-content: center;
  gap: 1.2rem;
  /* reduced gap */
}

.navbar li {
  font-size: 1rem;
  /* smaller links */
  cursor: pointer;
  position: relative;
  font-weight: 500;
}

.navbar li::after {
  content: "";
  position: absolute;
  left: 0;
  bottom: -4px;
  width: 0%;
  height: 2px;
  background: var(--color-accent);
  transition: width var(--transition-speed);
}

.navbar li:hover::after {
  width: 100%;
}

.navbar-theme-toggle {
  width: 35px;
  /* smaller icon */
  cursor: pointer;
  margin-left: 1rem;
  transition: transform var(--transition-speed);
}

.navbar-theme-toggle:hover {
  transform: rotate(15deg) scale(1.05);
}

/* Search Box */
.search-box {
  display: flex;
  align-items: center;
  background-color: var(--color-search-bg);
  padding: 6px 12px;
  /* smaller padding */
  border-radius: 20px;
  /* slimmer pill */
  transition: background-color var(--transition-speed);
}

.search-box input {
  padding: 4px;
  background: transparent;
  border: none;
  outline: none;
  color: var(--color-search-text);
  font-size: 0.9rem;
  max-width: 160px;
}

.search-box input::placeholder {
  color: var(--color-search-text);
  opacity: 0.7;
}

.search-box img {
  width: 18px;
  cursor: pointer;
}

.right {
  display: flex;
}

/* Responsive */
@media (max-width: 768px) {
  .navbar ul {
    display: none;
  }

  .search-box {
    max-width: 140px;
    padding: 5px 10px;
  }

  .navbar h1 {
    font-size: 1.2rem;
  }
}
