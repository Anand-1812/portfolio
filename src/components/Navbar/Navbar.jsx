// Navbar.jsx
import './Navbar.css';
import searchIconLight from '../../assets/search-w.png';
import searchIconDark from '../../assets/search-b.png';
import toggleLight from '../../assets/night.png';
import toggleDark from '../../assets/day.png';

function Navbar({ theme, setTheme }) {
  const toggleMode = () => {
    const newTheme = theme === 'dark' ? 'light' : 'dark';
    setTheme(newTheme);
  };

  const searchIcon = theme === 'dark' ? searchIconDark : searchIconLight;
  const toggleIcon = theme === 'dark' ? toggleDark : toggleLight;

  return (
    <div className="navbar">
      <h1>Anand Kumar</h1>
      <ul>
        <li>Home</li>
        <li>Projects</li>
        <li>About</li>
      </ul>
      <div className="search-box">
        <input type="text" placeholder="Search" />
        <img src={searchIcon} alt="Search Icon" />
      </div>
      <img
        src={toggleIcon}
        alt="Toggle Theme"
        className="navbar-theme-toggle"
        data-testid="theme-toggle"
        onClick={toggleMode}
      />
    </div>
  );
}

export default Navbar;

