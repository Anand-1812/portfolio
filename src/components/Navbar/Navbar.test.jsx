// Navbar.test.jsx
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, it, expect, vi } from 'vitest';
import Navbar from './Navbar';

describe('Navbar Component', () => {
  it('toggles from light to dark theme when clicked', async () => {
    const user = userEvent.setup();
    const setThemeMock = vi.fn();

    render(<Navbar theme="light" setTheme={setThemeMock} />);

    const toggleButton = screen.getAllByTestId('theme-toggle')[0];
    await user.click(toggleButton);

    expect(setThemeMock).toHaveBeenCalledTimes(1);
    expect(setThemeMock).toHaveBeenCalledWith('dark');
  });

  it('toggles from dark to light theme when clicked', async () => {
    const user = userEvent.setup();
    const setThemeMock = vi.fn();

    render(<Navbar theme="dark" setTheme={setThemeMock} />);

    const toggleButton = screen.getAllByTestId('theme-toggle')[0];
    await user.click(toggleButton);

    expect(setThemeMock).toHaveBeenCalledTimes(1);
    expect(setThemeMock).toHaveBeenCalledWith('light');
  });
});

