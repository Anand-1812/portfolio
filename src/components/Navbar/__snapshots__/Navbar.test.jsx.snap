// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Theme Toggle > renders correctly in dark mode 1`] = `
<div>
  <div
    class="navbar"
  >
    <h1>
      Anand Kumar
    </h1>
    <ul>
      <li>
        Home
      </li>
      <li>
        Projects
      </li>
      <li>
        About
      </li>
    </ul>
    <div
      class="search-box"
    >
      <input
        placeholder="Search"
        type="text"
      />
      <img
        alt=""
        src="/src/assets/search-b.png"
      />
    </div>
    <img
      alt="Toggle Theme"
      class="navbar-theme-toggle"
      data-testid="theme-toggle"
      src="/src/assets/day.png"
    />
  </div>
</div>
`;

exports[`Navbar Theme Toggle > renders correctly in light mode 1`] = `
<div>
  <div
    class="navbar"
  >
    <h1>
      Anand Kumar
    </h1>
    <ul>
      <li>
        Home
      </li>
      <li>
        Projects
      </li>
      <li>
        About
      </li>
    </ul>
    <div
      class="search-box"
    >
      <input
        placeholder="Search"
        type="text"
      />
      <img
        alt=""
        src="/src/assets/search-w.png"
      />
    </div>
    <img
      alt="Toggle Theme"
      class="navbar-theme-toggle"
      data-testid="theme-toggle"
      src="/src/assets/night.png"
    />
  </div>
</div>
`;
